src/core/public/application/scoped_history.ts

```diff
@@ -111,6 +115,25 @@ export class ScopedHistory<HistoryLocationState = unknown>
     return this.parentHistory.action;
   }
 
+  /**
+   * Fetches the parent location state
+   */
+  public fetchLocationState<T extends unknown = unknown>(): T {
+    this.verifyActive();
+    return cloneDeep(this.parentLocationState) as T;
+  }
+
+  /**
+   * Removes a key from the parent location state
+   *
+   * * @param key, the key to remove from the parent state
+   */
+  public removeFromLocationState(keys: string[]) {
+    this.verifyActive();
+    keys.forEach((key: string) => delete this.parentLocationState[key]);
+    this.replace(this.location, (this.parentLocationState as unknown) as HistoryLocationState);
+  }
```

Comment:
I want to keep wrapper class as thin as possible in order to avoid any bugs or unintended  discrepancies between ScopedHistory and raw History behavior.

This appears to be a utility that could be implemented elsewhere?

Reply:
I was able to remove this change as well, due to deleting directly from `ScopedHistory.location.state` working properly.